pkg/streamingpromql/testdata/ours/binary_operators.test

```diff
@@ -156,3 +156,113 @@ eval range from 0 to 2m step 1m single_series - many_series
 
 eval range from 0 to 2m step 1m many_series - single_series
   {env="prod"} -60 -150 -240
+
+clear
+
+# Binary operations on native histograms
+load 5m
+  first_histogram{job="test"}    {{schema:0 sum:5 count:4 buckets:[1 2 1]}}
+  second_histogram{job="test"}   {{schema:0 sum:10 count:6 buckets:[1 2 1]}}
+  third_histogram{job="control"} {{schema:0 sum:15 count:5 buckets:[3 4 2]}}
+  metric{job="test"}             2
+
+eval instant at 5m first_histogram + second_histogram
+  {job="test"} {{schema:0 sum:15 count:10 buckets:[2 4 2]}}
+
+eval instant at 5m second_histogram - first_histogram
+  {job="test"} {{schema:0 sum:5 count:2 buckets:[0 0 0]}}
+
+# Two histograms multiplied together result in a float 0
+eval instant at 5m first_histogram * second_histogram
+  {job="test"} 0
+
+# Cannot divide a histogram by a histogram
+eval instant at 5m first_histogram / second_histogram
+  {job="test"} NaN
+
+# Histogram multiplied by float
```

Comment:
It'd be good to add some test cases for things like `first_histogram + metric`, `first_histogram - metric` and `first_histogram atan2 metric`